File: src/transformers/features/protobuf/ProtobufCreateAssertEncodeTransformer.ts
import { ProtobufAssertEncodeProgrammer } from "../../../programmers/protobuf/ProtobufAssertEncodeProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateAssertEncodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "protobuf.createAssertEncode",
      write: ProtobufAssertEncodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufCreateAssertEncodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufCreateAssertEncodeTransformer = protobufCreateAssertEncodeTransformerNamespace{}

type protobufCreateAssertEncodeTransformerNamespace struct{}

func (protobufCreateAssertEncodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufAssertEncodeProgrammer.Write(call.TypeText)
}
